src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -6547,4 +6547,32 @@ internal sealed partial class LazyTranslatedQueryExpression : BaseTranslatedQuer
         }
         protected override IOperation ExpressionImpl => _lazyExpression.Value;
     }
+
+    internal sealed partial class FlowCapture : Operation, IFlowCaptureOperation
+    {
+        public FlowCapture(int id, SyntaxNode syntax, bool isInitialization, ITypeSymbol type, Optional<object> constantValue) :
+            base(OperationKind.FlowCapture, semanticModel:null, syntax, type, constantValue, isImplicit:true)
+        {
+            Id = id;
+            IsInitialization = isInitialization;
+        }
+
+        public int Id { get; }
+        public bool IsInitialization { get; }
+        public override IEnumerable<IOperation> Children
+        {
+            get
+            {
+                yield break;
```

Comment:
Consider returning `ImmutableArray<IOperation>.Empty`.

Reply:
> Consider returning ImmutableArray<IOperation>.Empty

What would be the motivation?

Reply:
Two motivations: 
(1) The code the is produced is much simpler. An iterator method with `yield break` produces the IL shown below.
(2) When debugging, you can see the result without stepping through code. 

```
  // Nested Types
    .class nested private auto ansi sealed beforefieldinit '<M>d__1'
        extends [mscorlib]System.Object
        implements class [mscorlib]System.Collections.Generic.IEnumerable`1<int32>,
                   [mscorlib]System.Collections.IEnumerable,
                   class [mscorlib]System.Collections.Generic.IEnumerator`1<int32>,
                   [mscorlib]System.IDisposable,
                   [mscorlib]System.Collections.IEnumerator
    {
        .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
            01 00 00 00
        )
        // Fields
        .field private int32 '<>1__state'
        .field private int32 '<>2__current'
        .field private int32 '<>l__initialThreadId'

        // Methods
        .method public hidebysig specialname rtspecialname 
            instance void .ctor (
                int32 '<>1__state'
            ) cil managed 
        {
            .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = (
                01 00 00 00
            )
            // Method begins at RVA 0x2063
            // Code size 25 (0x19)
            .maxstack 8

            IL_0000: ldarg.0
            IL_0001: call instance void [mscorlib]System.Object::.ctor()
            IL_0006: ldarg.0
            IL_0007: ldarg.1
            IL_0008: stfld int32 Program/'<M>d__1'::'<>1__state'
            IL_000d: ldarg.0
            IL_000e: call int32 [mscorlib]System.Environment::get_CurrentManagedThreadId()
            IL_0013: stfld int32 Program/'<M>d__1'::'<>l__initialThreadId'
            IL_0018: ret
        } // end of method '<M>d__1'::.ctor

        .method private final hidebysig newslot virtual 
            instance void System.IDisposable.Dispose () cil managed 
        {
            .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = (
                01 00 00 00
            )
            .override method instance void [mscorlib]System.IDisposable::Dispose()
            // Method begins at RVA 0x2050
            // Code size 1 (0x1)
            .maxstack 8

            IL_0000: ret
        } // end of method '<M>d__1'::System.IDisposable.Dispose

        .method private final hidebysig newslot virtual 
            instance bool MoveNext () cil managed 
        {
            .override method instance bool [mscorlib]System.Collections.IEnumerator::MoveNext()
            // Method begins at RVA 0x2080
            // Code size 21 (0x15)
            .maxstack 2
            .locals init (
                [0] int32
            )

            IL_0000: ldarg.0
            IL_0001: ldfld int32 Program/'<M>d__1'::'<>1__state'
            IL_0006: stloc.0
            IL_0007: ldloc.0
            IL_0008: brfalse.s IL_000c
            IL_000a: ldc.i4.0
            IL_000b: ret
            IL_000c: ldarg.0
            IL_000d: ldc.i4.m1
            IL_000e: stfld int32 Program/'<M>d__1'::'<>1__state'
            IL_0013: ldc.i4.0
            IL_0014: ret
        } // end of method '<M>d__1'::MoveNext

        .method private final hidebysig specialname newslot virtual 
            instance int32 'System.Collections.Generic.IEnumerator<System.Int32>.get_Current' () cil managed 
        {
            .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = (
                01 00 00 00
            )
            .override method instance !0 class [mscorlib]System.Collections.Generic.IEnumerator`1<int32>::get_Current()
            // Method begins at RVA 0x20a1
            // Code size 7 (0x7)
            .maxstack 8

            IL_0000: ldarg.0
            IL_0001: ldfld int32 Program/'<M>d__1'::'<>2__current'
            IL_0006: ret
        } // end of method '<M>d__1'::'System.Collections.Generic.IEnumerator<System.Int32>.get_Current'

        .method private final hidebysig newslot virtual 
            instance void System.Collections.IEnumerator.Reset () cil managed 
        {
            .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = (
                01 00 00 00
            )
            .override method instance void [mscorlib]System.Collections.IEnumerator::Reset()
            // Method begins at RVA 0x20a9
            // Code size 6 (0x6)
            .maxstack 8

            IL_0000: newobj instance void [mscorlib]System.NotSupportedException::.ctor()
            IL_0005: throw
        } // end of method '<M>d__1'::System.Collections.IEnumerator.Reset

        .method private final hidebysig specialname newslot virtual 
            instance object System.Collections.IEnumerator.get_Current () cil managed 
        {
            .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = (
                01 00 00 00
            )
            .override method instance object [mscorlib]System.Collections.IEnumerator::get_Current()
            // Method begins at RVA 0x20b0
            // Code size 12 (0xc)
            .maxstack 8

            IL_0000: ldarg.0
            IL_0001: ldfld int32 Program/'<M>d__1'::'<>2__current'
            IL_0006: box [mscorlib]System.Int32
            IL_000b: ret
        } // end of method '<M>d__1'::System.Collections.IEnumerator.get_Current

        .method private final hidebysig newslot virtual 
            instance class [mscorlib]System.Collections.Generic.IEnumerator`1<int32> 'System.Collections.Generic.IEnumerable<System.Int32>.GetEnumerator' () cil managed 
        {
            .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = (
                01 00 00 00
            )
            .override method instance class [mscorlib]System.Collections.Generic.IEnumerator`1<!0> class [mscorlib]System.Collections.Generic.IEnumerable`1<int32>::GetEnumerator()
            // Method begins at RVA 0x20c0
            // Code size 43 (0x2b)
            .maxstack 2
            .locals init (
                [0] class Program/'<M>d__1'
            )

            IL_0000: ldarg.0
            IL_0001: ldfld int32 Program/'<M>d__1'::'<>1__state'
            IL_0006: ldc.i4.s -2
            IL_0008: bne.un.s IL_0022
            IL_000a: ldarg.0
            IL_000b: ldfld int32 Program/'<M>d__1'::'<>l__initialThreadId'
            IL_0010: call int32 [mscorlib]System.Environment::get_CurrentManagedThreadId()
            IL_0015: bne.un.s IL_0022
            IL_0017: ldarg.0
            IL_0018: ldc.i4.0
            IL_0019: stfld int32 Program/'<M>d__1'::'<>1__state'
            IL_001e: ldarg.0
            IL_001f: stloc.0
            IL_0020: br.s IL_0029
            IL_0022: ldc.i4.0
            IL_0023: newobj instance void Program/'<M>d__1'::.ctor(int32)
            IL_0028: stloc.0
            IL_0029: ldloc.0
            IL_002a: ret
        } // end of method '<M>d__1'::'System.Collections.Generic.IEnumerable<System.Int32>.GetEnumerator'

        .method private final hidebysig newslot virtual 
            instance class [mscorlib]System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator () cil managed 
        {
            .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = (
                01 00 00 00
            )
            .override method instance class [mscorlib]System.Collections.IEnumerator [mscorlib]System.Collections.IEnumerable::GetEnumerator()
            // Method begins at RVA 0x20f7
            // Code size 7 (0x7)
            .maxstack 8

            IL_0000: ldarg.0
            IL_0001: call instance class [mscorlib]System.Collections.Generic.IEnumerator`1<int32> Program/'<M>d__1'::'System.Collections.Generic.IEnumerable<System.Int32>.GetEnumerator'()
            IL_0006: ret
        } // end of method '<M>d__1'::System.Collections.IEnumerable.GetEnumerator

        // Properties
        .property instance int32 'System.Collections.Generic.IEnumerator<System.Int32>.Current'()
        {
            .get instance int32 Program/'<M>d__1'::'System.Collections.Generic.IEnumerator<System.Int32>.get_Current'()
        }
        .property instance object System.Collections.IEnumerator.Current()
        {
            .get instance object Program/'<M>d__1'::System.Collections.IEnumerator.get_Current()
        }

    } // end of class <M>d__1


    .method public hidebysig 
        instance class [mscorlib]System.Collections.Generic.IEnumerable`1<int32> M () cil managed 
    {
        .custom instance void [mscorlib]System.Runtime.CompilerServices.IteratorStateMachineAttribute::.ctor(class [mscorlib]System.Type) = (
            01 00 0f 50 72 6f 67 72 61 6d 2b 3c 4d 3e 64 5f
            5f 31 00 00
        )
        // Method begins at RVA 0x2052
        // Code size 8 (0x8)
        .maxstack 8

        IL_0000: ldc.i4.s -2
        IL_0002: newobj instance void Program/'<M>d__1'::.ctor(int32)
        IL_0007: ret
    } // end of method Program::M
```

Reply:
There are a lot more instances of `yield break` in the nodes, this seems a bit of scope for this PR. File a followup bug?

---
In reply to: [162142374](https://github.com/dotnet/roslyn/pull/24263#discussion_r162142374) [](ancestors = 162142374)